### PR TITLE
[CAL-1391] Return deal_id and milestone_id from tasks.info and tasks.list

### PR DESCRIPTION
**Pull request by r1dotl**
TICKET: https://teamleader.atlassian.net/browse/CAL-1391

Allow the tasks.list and tasks.info to return the milestone_id and deal_id.

If the milestone_id/deal_id don't exist it will return null, but the attribute will be always returned.

### CODE REVIEW

File: src/08-tasks/tasks.apib [hunk 1]
```diff
@@ -48,6 +48,8 @@ Get a list of tasks.
                             + user
                             + team
                     + id: `f29abf48-337d-44b4-aad4-585f5277a456` (string)
+                + milestone_id: `6fac0bf0-e803-424e-af67-76863a3d7d16` (string, nullable)
```
Comment (psrpinto) [inline src/08-tasks/tasks.apib]: B: Should be a `milestone` object instead of `milestone_id`. Same for `deal` and same for `.info`.

**Review by psrpinto — CHANGES_REQUESTED** [review]

File: src/changes-backwards-compatible.apib [hunk 2]
```diff
@@ -5,7 +5,7 @@ We list all backwards-compatible additions here. These are currently available i
 (There is also a [list of backwards-incompatible upgrades](#changelog) available, but those only apply if you [upgrade your API version](#upgrading-your-api-version).)
 
 #### Latest
-
+- `tasks.list` and `tasks.info` now returns the milestone and deal objects.
```
Comment (psrpinto) [inline src/changes-backwards-compatible.apib]:
```suggestion
- `tasks.list` and `tasks.info` now return the milestone and deal objects.
```

File: apiary.apib [hunk 3]
```diff
@@ -3315,6 +3329,8 @@ Create a new task.
         + description (string, required)
         + due_at: `2016-02-04T16:44:33+00:00` (string, required)
         + work_type_id: `32665afd-1818-0ed3-9e18-a603a3a21b95` (string, required)
+        + milestone_id: `32665afd-1818-0ed3-9e18-a603a3a21b95` (string, optional)
+        + deal_id: `32665afd-1818-0ed3-9e18-a603a3a21b95` (string, optional)
```
Comment (psrpinto) [inline apiary.apib]: B: These changes are not from this PR, right? Same for the whitespace changes.

**Review by psrpinto — CHANGES_REQUESTED** [review]

**Review by mikeverf — APPROVED** [review]

**Review by psrpinto — APPROVED** [review]
I think it would have been better to figure out why the unwanted changes were here and act accordingly, instead of removing them through extra commits. Also, the whitespace changes are still there.

**Review by JorenSaey — APPROVED** [review]

**Review by jenssegers — APPROVED** [review]

**Comment by stivni** [comment]
> I think it would have been better to figure out why the unwanted changes were here and act accordingly, instead of removing them through extra commits.

Indeed, because if these commits get to master, those lines will be removed if they are in master...

**Review by stivni — COMMENTED** [review]
I wonder if we need to put this both as a top level property...
Have you considered other options, like a "links" section?
Not sure if that makes more sense here, but want to consider it for a second...

**Comment by r1dotl** [comment]
This is already a mess, so i will close this one and open a new one clean.

**Comment by r1dotl** [comment]
Creating a new one more clean.

**Pull request closed without merging**